### DISCUSSION TITLE
Clarification about grid generation units

### DISCUSSION THREAD

**Discussion by tm-jc-nacpil** [thinkingmachines/geowrangler#52] [Q&A] [answered]
Hi! I'm currently using the [grid generation tutorial](https://geowrangler.web.app/tutorial.grids.html) and would like to clarify if the output grids in the example are **in degrees or meters**?

In the create grids section, the tutorial mentions that the units of the grids are dependent on the units of the projection.

> Create a grid generator with a size of 50000. The units of the grid size are dependent on the projection of the geodataframe, in this case, EPSG:4326.

When printing out the CRS of `region3_gdf`, it shows that the units are in decimal degrees. 

```python
>> region3_gdf.crs  # CRS info
<Geographic 2D CRS: EPSG:4326>
Name: WGS 84
Axis Info [ellipsoidal]:
- Lat[north]: Geodetic latitude (degree)
- Lon[east]: Geodetic longitude (degree)
Area of Use:
- name: World.
- bounds: (-180.0, -90.0, 180.0, 90.0)
Datum: World Geodetic System 1984 ensemble
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich
```
However, the output grids in the tutorial are described to be in kilometer scale:
```python
grid_generator5k = grids.SquareGridGenerator(
    region3_gdf, 5000
)  # 5 km x 5 km square cells
```

Let me know also if this is the correct place for this question or if I should place this in Discussions instead. Thanks! :D

**Comment by ghost** [comment]
Hi, JC. Thanks for sending in your clarification. I'm converting this issue into a discussion for better classification and to better document the responses to this question.

**Comment by butchtm** [comment, marked as answer]
Hi @tm-jc-nacpil
The `SquareGridGenerator`  has a parameter called `grid_projection` with the default value `EPSG:3857` which is a [CRS with a unit of 'meters'](https://epsg.io/3857). The gdf is [ reprojected before the grids are computed](https://github.com/thinkingmachines/geowrangler/blob/b62f0aced4dd37616237d3464b13055fb2c1920c/geowrangler/grids.py#L88), which is why the cell_size is `5000 (meters)`  or `5 (km)` . 

In effect,  this is the call with the default parameters included
```
grid_generator5k = grids.SquareGridGenerator(
    region3_gdf, 
   cell_size=5000,
   grid_projection = 'EPSG:3857',
  boundary=None
)  # 5 km x 5 km square cells
```
As a note to improve the tutorial, maybe we should create a PR for the grids tutorial to clarify this? cc @tm-kah-alforja

**Reply by tm-jc-nacpil** [reply]
Thanks for clearing this up :D Is the output grids also in the specified `grid_projection` or are they reprojected back in the original gdf's crs?

**Reply by jtmiclat** [reply]
@tm-jc-nacpil they are reprojected back to the original gdf's crs
https://github.com/thinkingmachines/geowrangler/blob/5d4bc332ecd40b386aa997f59b9da461a7e7fd1f/geowrangler/grids.py#L118
